src/libstore/binary-cache-store.cc

```diff
@@ -414,15 +424,21 @@ StorePath BinaryCacheStore::addToStore(
     });
     return addToStoreCommon(*source, repair, CheckSigs, [&](HashResult nar) {
         ValidPathInfo info {
-            makeFixedOutputPath(method, h, name, references),
+            *this,
+            name,
+            FixedOutputInfo {
+                .hash = {
+                    .method = method,
+                    .hash = h,
+                },
+                .references = {
+                    .others = references,
+                    .self = false,
```

Comment:
```suggestion
                     // caller is not capable of creating a self-reference, because this is content-addressed without modulus
                    .self = false,
```